### ISSUE TITLE
MaterialTextBox wrongly sets invalid style on some browsers during event cycle

### ISSUE THREAD

**Issue by esoco** [GwtMaterialDesign/gwt-material#424]
On certain event handling cycles (e.g. key press - disable - enable) the MaterialTextBox will add the the class "invalid" to it's style name even if no validation has been activated. This will render the text box with an error style. And this happen only on certain browsers. In my tests that have been Chrome, Vivaldi (probably all Webkit based browsers) and IE 11. It doesn't occur on Firefox and Edge. I have been able to reproduce the problem with the simple code snippet below. There it may not occur on every key press, but in other constellations with longer delays until the re-enabling it will occur predictably.

The strange thing is that I cannot find the cause for this style change. As far as I can see the only place where the style name "invalid" is set is in the method `setError()` of `MaterialValueBox`. But this method is not called if I execute the example (or my original code that causes this problem). Can it be that the mentioned browsers somehow set this name on the element and that this is only a conflict with the pre-set CSS for "invalid"? But then the question is why the browser considers the content of the field invalid. Or is there some hidden GWT "magic" at work?

```java
final MaterialTextBox textbox = new MaterialTextBox();

panel.add(textbox);

textbox.addKeyUpHandler(new KeyUpHandler()
{
	public void onKeyUp(KeyUpEvent rEvent)
	{
		textbox.setEnabled(false);
		Scheduler.get().scheduleDeferred(new ScheduledCommand()
		{
			public void execute()
			{
				textbox.setEnabled(true);
			}
		});
	}
});

```

**Comment by esoco** [comment]
As I just found out the problem is caused by the style name "validate" that `MaterialValueBox` adds to it's input widget in the `setType()` method. It seems that this style name triggers some internal validation mechanism of Webkit and IE11 which causes the "invalid" style to appear (unless I missed something in gwt-material). After I removed the style name the invalid style no longer appears on input.

**Comment by kevzlou7979** [comment]
I tried to reproduce your issue with same code as yours, but the issue doesn't exist. Also when I checked on setType() theres no appending of classes to input widget (Check here for 1.6.x https://github.com/GwtMaterialDesign/gwt-material/blob/release_1.6.2/gwt-material/src/main/java/gwt/material/design/client/ui/MaterialValueBox.java#L206) .

If you a simple project that we can test it on, that will be great.
On question have you implement some validation e.g ``` allowBlank="false" ``` if yes that will trigger the ``` validate ``` stylename into the valuebox.
